qupulse/utils/__init__.py

```diff
@@ -122,3 +125,32 @@ def forced_hash(obj) -> int:
             return hash(tuple(map(forced_hash, obj)))
 
         raise
+
+
+def to_next_multiple(sample_rate: ExpressionLike, quantum: int,
+                     min_quanta: Optional[int] = None) -> Callable[[ExpressionLike],ExpressionScalar]:
+    """Construct a helper function to expand a duration to one corresponding to
+    valid sample multiples according to the arguments given.
+    Useful e.g. for PulseTemplate.pad_to's 'to_new_duration'-argument.
+
+    Args:
+        sample_rate: sample rate with respect to which the duration is evaluated.
+        quantum: number of samples to whose next integer multiple the duration shall be rounded up to.
+        min_quanta: number of multiples of quantum not to fall short of.
+    Returns:
+        A function that takes a duration (ExpressionLike) as input, and returns
+        a duration rounded up to the next valid samples count in given sample rate.
+        The function returns 0 if duration==0, <0 is not checked if min_quanta is None.
+
+    """
+    sample_rate = ExpressionScalar(sample_rate)
+    #is it more efficient to omit the Max call if not necessary?
+    if min_quanta is None:
+        #double negative for ceil division.
+        return lambda duration: -(-(duration*sample_rate)//quantum) * (quantum/sample_rate)
+    else:
+        #still return 0 if duration==0
+        return lambda duration: ExpressionScalar((quantum/sample_rate)\
+                                    *Max(-(-(ExpressionScalar(duration)*sample_rate)//quantum),min_quanta)\
```

Comment:
We should try to avoid relying on sympy internals.

Reply:
Somehow a functionality that evaluates the min_quanta within the Expression has to be implemented. I did not directly see a different way than to use some sympy-functionality within the expression. Or are you just referring to the fact that this should be written out in e.g. a string to initialize the ExpressionScalar with?

Reply:
`f'max({min_quanta}, {ceiled_division(duration, quantum)}) * {quantum}'` could work although I am not sure how robust the interaction with TimeType is.

Reply:
This seems to indeed be a bit problematic with TimeType (the tests give me weird behavior at the moment with a similar string), although I do not fully understand why.

Reply:
7c520a7 might resolve this; at least the test cases seem to work